src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = this.random.nextInt(FigureTypes.values().length);
+        FigureTypes figureType = FigureTypes.values()[index];
+        String color = this.colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case CIRCLE:
+                return new Circle(
+                        color,
+                        this.random.nextDouble()
+                );
```

Comment:
create a private method to create each of figure and invoke it here

Reply:
Done